Другой вариант парсера

Crazzzzy opened the pull request:
## Описание

Я выкладываю свою версию парсера, который парсит названия книг и авторов с сайтов livelib, readrate, libs.
Так же можно сравнивать топы разных сайтов и находить совпадения.
Расписывать я не умею, просто решил поделиться своим вариантом.
Критика приветсвуется :3
## Тип изменений
- [X] Домашнее задание (если выбрал этот пункт, удали раздел "Причины")
- [ ] Исправление ошибки (bug fix) (некритическое изменение, которое исправляет что-то, обсуждаемое в issues)
- [ ] Новая фича (new feature) (некритическое изменение, добавляющее функциональности)
- [X] Критическое изменение (breaking change) (исправление или фича, которая сильно меняет структуру/функциональность кода)
## Чеклист:
- [x] Мой код соответствует стилю, в котором написан весь проект.
- [X] Мои изменения требуют изменений в документации.
- [ ] Я внес соответствующие правки в документацию.
- [ ] Я добавил тесты, проверяющие мой код.
- [x] Все новые и существующие тесты успешно пройдены.

Crazzzzy committed:
Create main.py

Crazzzzy committed:
Create sites.py

Crazzzzy committed:
Create search_matches.py

murych commented on my_parser/sites.py:
```diff
@@ -0,0 +1,67 @@
+	def __init__(self, url):
+		self.url = url
+		# Из полного адреса выбираем имя.домен
+		self.name = re.findall('(\w+\.\w+)\/', url)[0]
```
я готов был убивать за регексп, который позволил бы мне вынуть домен. как, КАК вы это делаете?

murych commented on my_parser/sites.py:
```diff
@@ -0,0 +1,67 @@
+			author = i.find('a')
+			if author != None:
+				top_libs_authors.append(author)
+		# (Костыль) У книги на 25 месте нету автора
```
готов поклясться, что видел автора у книги на 25 месте:
Муравей в стеклянной банке. Чеченские дневники 1994-2004 гг.
Автор(ы): Жеребцова Полина

murych committed:
обработка сайта Libs.com

надо провеить, но автор прятался в теге <a href="/a/....">. в связи с этим - убрал "костыль"
lambda-frela/parser#10

murych committed:
reverted to be7f58fb444fbf2322f27aa59b696129dec733ed

lambda-frela/pasrer#10

murych committed:
исправил свои заблуждения в a7f75c41a9746d0bd7f7085c6863fc15ff93c54a

- "автор" отсутствует в книге №27
- исправил исключение в main.py
- упростил класс Libs
lambda-frela/parser#10

murych committed:
поправил код

lambda-frela/parser#10

murych committed:
реструктурировал проект

lambda-frela/parser#10

Crazzzzy committed:
Merge pull request #1 from murych/object_oriented

обработка сайта Libs.com
